providers/implementations/rands/drbg.c

```diff
@@ -62,12 +66,18 @@ void ossl_drbg_unlock(void *vctx)
         CRYPTO_THREAD_unlock(drbg->lock);
 }
 
-static int ossl_drbg_lock_parent(PROV_DRBG *drbg)
+static int ossl_drbg_lock_parent(PROV_DRBG *drbg, int read)
 {
     void *parent = drbg->parent;
 
-    if (parent != NULL
-            && drbg->parent_lock != NULL
+    if (parent == NULL)
+        return 1;
+    if (drbg->parent_lock_ex != NULL
+            && !drbg->parent_lock_ex(parent, read)) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_PARENT_LOCKING_NOT_ENABLED);
+        return 0;
+    }
+    if (drbg->parent_lock != NULL
```

Comment:
What if both are implemented?

Reply:
Then we should use `parent_lock_ex` in preference (which is what this code does).

Reply:
Ah...but now I see the problem! Fix coming up.

Reply:
Fixed.